pipelines/nextflow/subworkflows/prepare_metadata.nf

```diff
@@ -0,0 +1,50 @@
+/*
+See the NOTICE file distributed with this work for additional information
+regarding copyright ownership.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+nextflow.enable.dsl=2
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    IMPORT LOCAL MODULES/FUNCTIONS
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+// Utilities
+include { index_metadata } from '../modules/utils.nf'
+// Module
+include { QUERY_CORE_META } from '../modules/query_core_meta.nf'
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    RUN MAIN WORKFLOW
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+workflow PREPARE_METADATA{
+    // Generate a meta object from a database meta table and dump in JSON format
+    take:
+        input_cores
+        core_metakeys // input meta keys JSON
+
+    main:
+        QUERY_CORE_META(input_cores, core_metakeys)
```

Comment:
why the output of QUERY_CORE_METADATA needs an extra step? Can the output be emitted directly by the module itself?

Reply:
The setup of this was originally a module, but I've opted for a subworkflow as the combined module processing of `QUERY_CORE_META` followed by a function `index_metadata` to map the output meta into a named meta_map is much more aligned to subworkflow than a module alone.